main.py

```
@@ -1,69 +1,465 @@
+                self.save_markdown(markdown_content, output_file)
+                
+                # Create and save chunks if enabled
+                if save_chunks:
+                    chunks = self.create_chunks(markdown_content, url)
+                    # Create URL-specific chunk directory to prevent filename collisions
+                    url_chunk_dir = f"{chunk_dir}/{filename.replace('.md', '')}"
+                    self.save_chunks(chunks, url_chunk_dir, chunk_format)
+                
+                successfully_scraped.append(url)
+                
+            except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)